packages/database/src/Builder/QueryBuilders/CountQueryBuilder.php

```diff
@@ -10,14 +10,15 @@
 use Tempest\Database\Query;
 use Tempest\Database\QueryStatements\CountStatement;
 use Tempest\Database\QueryStatements\WhereStatement;
+use Tempest\Database\UsesDatabase;
 use Tempest\Support\Conditions\HasConditions;
 
 /**
  * @template TModelClass of object
  */
 final class CountQueryBuilder implements BuildsQuery
 {
-    use HasConditions;
+    use HasConditions, UsesDatabase;
```

Comment:
I'm not sure whether `useDatabase` is the best name. Other alternatives I could think of:

- `inDatabase`
- `withDatabase`
- `withConnection`
- `useConnection`

I'm ok changing it if others have a strong opinion

Reply:
What about `onDatabase`, as in "execute this query on this database"?

Reply:
I'm fine with that!